### PR TITLE
Add Ability to Resize GUI Window

### PR DESCRIPTION
**Pull request by Cvarier**
- Made window resizable
- Adjusted fonts and line styles for better visibility upon window resize
- Verified working on following OSes
  - WSL 2 - Ubuntu 20.04.1 (Focal)
  - Native Windows
  - Native Linux
 
image

image

### CODE REVIEW

File: src/gui_tool/gui/general_gui.py
Comment (Cvarier) [inline src/gui_tool/gui/general_gui.py:72]: Adding the flags `cv2.WINDOW_NORMAL | cv2.WINDOW_GUI_NORMAL` gives the ability to resize the window while correctly rescaling user input.

**Comment by Ercadio** [comment]
How does this affect bounding boxes? I think that you are missing some work to update bounding boxes accordingly

File: src/gui_tool/gui/general_gui.py
Comment (tonymeng1) [inline src/gui_tool/gui/general_gui.py:74]: Is there any specific reason why this needs to be changed?

**Review by tonymeng1 — APPROVED** [review]
Lol Actually works, at least on my windows machine. Didn't know it was build in... Nice find XD

Maybe wait on feedback from other OS's before merging though

**Review by tonymeng1 — APPROVED** [review]
Lol Actually works, at least on my windows machine. Didn't know it was build in... Nice find XD

Maybe wait on feedback from other OS's before merging though

**Comment by Cvarier** [comment]
> How does this affect bounding boxes? I think that you are missing some work to update bounding boxes accordingly

@Ercadio, I just verified that the bb maps correctly when resizing. Please see if this works for you though on your OS.

**Comment by Cvarier** [comment]
> Lol Actually works, at least on my windows machine. Didn't know it was build in... Nice find XD
> 
> Maybe wait on feedback from other OS's before merging though

@tonymeng1 How are you running this on Windows exactly? Is it native or through WSL?

**Comment by Cvarier** [comment]
Can you verify if this PR works for you @nicholas-geofroy (on Linux) @jackwilliamson (on Mac)

**Comment by tonymeng1** [comment]
I'm running on native windows. 

There's a pretty good chance it works on other OS's too since you're using a library functionality, but I think the most important thing GUI-wise right now is not corrupting the dataset

**Review by nicholas-geofroy — APPROVED** [review]

File: src/gui_tool/gui/bb/BoundingBoxManager.py
Comment (nicholas-geofroy) [inline src/gui_tool/gui/bb/BoundingBoxManager.py:11]: can we keep the thickness at 1? having thicker boxes makes it harder to tell where the real edge is

File: src/gui_tool/gui/sp_gui.py
Comment (nicholas-geofroy) [inline src/gui_tool/gui/sp_gui.py:290]: whats the difference here? I find the font is kinda blurry at least on linux
image

File: src/gui_tool/legacy/gui_managers.py
Comment (nicholas-geofroy) [inline src/gui_tool/legacy/gui_managers.py:183]: same comment as above

**Review by nicholas-geofroy — APPROVED** [review]
works on linux, and some minor comments

File: src/gui_tool/gui/bb/BoundingBoxManager.py
Comment (Cvarier) [reply src/gui_tool/gui/bb/BoundingBoxManager.py:11]: That's a good point, but if I leave it at 1, then some of the sides of the box flicker while animating between frames.

File: src/gui_tool/gui/sp_gui.py
Comment (Cvarier) [reply src/gui_tool/gui/sp_gui.py:290]: When resizing the window, the logger text will get jagged/aliased, making it harder to read since it's so thin. The only difference is that this font is thicker, and imo, easier to read.

File: src/gui_tool/gui/bb/BoundingBoxManager.py
Comment (nicholas-geofroy) [reply src/gui_tool/gui/bb/BoundingBoxManager.py:11]: oh okay than 2 is probably better, dont want a lot of visual flickering

**Pull request merged**
